### DISCUSSION TITLE
Undo does not work reliably with Table Editor enabled?

### DISCUSSION THREAD

**Discussion by mprewitt** [Zettlr/Zettlr#3635] [General]
I am using Zettlr version 2.3.0 for Windows.

### Using Undo With Table Editor Is Glitchy
I have the Table Editor active. Edits to table appear to be sometimes undo-able. It is sort of hit and miss though. 

- If I use the Table Editor UI to delete a row, and then undo, sometimes it works, sometimes it doesn't. 
- Occasionally undoing edits to the content will cause the table to render as text, and in those cases sometimes rows or columns that had been deleted, and then reverted with undo, but nevertheless did not reappear, are still there in the text view. Changing the content in those rows/columns will bring them back in the regular view.
- Sometimes edits to the content can be undone. Other times the app skips them and proceeds to undo other parts of the document. 

### Secondary issue
Not directly related, but sometimes inserting a table is glitchy too. It's my first day using the app, and I inserted a second table after the first one that I had been experimenting with. But instead of a nicely formatted table, I got this:

```
|    |    |    |    |
|    |    |    |    |
|    |    |    |    |
|    |    |    |    |
```

I can still get that same result in the same part of the document, but elsewhere in the document I can create tables normally. I originally thought the previous table had gotten broken during undo/redo tests, and this caused the next table inserted into the document to be malformed. But it appears that actually all the subsequent tables are malformed, but the subsequent ones were still displayed as tables.

**Comment by nathanlesage** [comment]
The TableEditor is independent of the main editor, meaning if you change a cell, to write the table contents back to the actual file (so that the file itself contains these changes) you have to focus a different cell. I'm currently pondering better ways of making this "write back changes from the visual table into the document" transparent, but these won't come until 3.0
